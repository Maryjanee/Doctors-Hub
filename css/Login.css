.logo {
  margin: 1rem auto;
}
/*# sourceMappingURL=Login.css.map */