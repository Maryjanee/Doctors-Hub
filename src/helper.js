/* eslint-disable import/prefer-default-export */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable arrow-body-style */
import { BrowserRouter } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import rootReducer from './reducers';

export const renderComponent = (Component, props = {}) => {
  return (
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  );
};

export const renderWithRedux = (component,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  } = {}) => ({
  ...render(
    <Provider store={store}>
      renderComponent(component)
    </Provider>,
  ),
});
