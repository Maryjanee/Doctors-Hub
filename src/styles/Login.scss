
.logo {
  padding: 1rem;
}
