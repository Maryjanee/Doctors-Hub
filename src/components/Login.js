/* eslint-disable react/no-this-in-sfc */
import { Link, Redirect, useHistory } from 'react-router-dom';
import { useState } from 'react';
import AuthService from '../services/auth-service';
import '../styles/Login.scss';

const INITIAL_LOGIN_FORM = {
  email: '',
  password: '',
};

const Login = () => {
  const [loginError, setLoginError] = useState(null);
  const [loginForm, updateLoginForm] = useState(INITIAL_LOGIN_FORM);
  const history = useHistory();

  const handleChange = event => {
    const { name, value } = event.target;
    updateLoginForm(prev => ({
      ...prev,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();
    const { email, password } = loginForm;
    updateLoginForm(INITIAL_LOGIN_FORM);
    if (email && password) {
      const response = await AuthService.login({ email, password });
      if (response && response.token) {
        localStorage.setItem('auth-token', response.token);
        setLoginError(null);
        history.push('/doctors');
      }
      if (!response || response.error) {
        setLoginError('Login failed. Please try again');
      }
    }
  };

  if (localStorage.getItem('auth-token')) {
    return <Redirect to="/doctors" />;
  }

  return (
    <div className="main-bg">
      <div className="container">
        <h1 className="logo">Doctors Hub</h1>
        <div className="form-container">
          <form onSubmit={handleSubmit} className="d-flex flex-cl">

            <input type="email" id="email" placeholder="Email" name="email" value={loginForm.email} onChange={handleChange} />

            <input type="password" id="password" name="password" placeholder="Password" value={loginForm.password} onChange={handleChange} />

            <input type="submit" value="Login" className="submit" />
          </form>

          <p className="text-center">
            Not Registered?
            <Link to="/signup" className="create">Create an Account</Link>
          </p>
          {loginError && <div>{loginError}</div>}
        </div>
      </div>
    </div>

  );
};
export default Login;
