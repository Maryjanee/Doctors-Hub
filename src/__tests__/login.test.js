import renderer from 'react-test-renderer';
import Login from '../components/Login';
import { renderComponent } from './helper';

describe('Login Page', () => {
  it('renders correctly', () => {
    const component = renderComponent(Login);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
