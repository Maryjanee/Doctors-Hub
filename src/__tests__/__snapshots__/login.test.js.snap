// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page deep snapshot renders correctly 1`] = `
<div
  className="main-bg"
>
  <div
    className="container"
  >
    <h1
      className="logo"
    >
      Doctors Hub
    </h1>
    <div
      className="form-container"
    >
      <form
        className="d-flex flex-cl"
        onSubmit={[Function]}
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Username"
          type="email"
          value=""
        />
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <input
          className="submit"
          type="submit"
          value="Login"
        />
      </form>
      <p
        className="text-center"
      >
        Not Registered?
        <a
          className="create"
          href="/signup"
          onClick={[Function]}
        >
          Create an Account
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`Login Page renders correctly 1`] = `
<div
  className="main-bg"
>
  <div
    className="container"
  >
    <h1
      className="logo"
    >
      Doctors Hub
    </h1>
    <div
      className="form-container"
    >
      <form
        className="d-flex flex-cl"
        onSubmit={[Function]}
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Username"
          type="email"
          value=""
        />
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <input
          className="submit"
          type="submit"
          value="Login"
        />
      </form>
      <p
        className="text-center"
      >
        Not Registered?
        <a
          className="create"
          href="/signup"
          onClick={[Function]}
        >
          Create an Account
        </a>
      </p>
    </div>
  </div>
</div>
`;
